{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "Purpose is to figure out why the raw sequences are messed up."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/usr/bin/env python\n",
    "#map_motif.py\n",
    "#Ciera Martinez\n",
    "\n",
    "import re\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os, sys\n",
    "from Bio import motifs\n",
    "from Bio import SeqIO \n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Alphabet import IUPAC, generic_dna, generic_protein\n",
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "## read in alignment and motif \n",
    "alignment = list(SeqIO.parse(\"../../montium_data/4_alignments/align_outlier_rm_with_length_VT11157.fa\", \"fasta\"))\n",
    "motif = motifs.read(open(\"../data/Dmel_pwms/bcd_FlyReg.fm\"), \"pfm\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "## Used later\n",
    "\n",
    "raw_sequences = []\n",
    "for record in alignment:\n",
    "    raw_sequences.append(SeqRecord(record.seq.ungap(\"-\"), id = record.id))\n",
    "\n",
    "## make raw sequences all IUPAC.IUPACUnambiguousDNA()\n",
    "raw_sequences_2 = []\n",
    "for seq in raw_sequences:\n",
    "    raw_sequences_2.append(Seq(str(seq.seq), IUPAC.IUPACUnambiguousDNA()))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        0      1      2      3\n",
      "A:   0.09   0.73   0.40   0.32\n",
      "C:   0.08   0.15   0.20   0.63\n",
      "G:   0.41   0.08   0.20   0.03\n",
      "T:   0.42   0.04   0.20   0.02\n",
      "\n",
      "3077\n",
      "3074\n",
      "[-6.1402273 -5.1402273 -0.930774  ... -4.1402273 -6.2016277 -5.1402273]\n"
     ]
    }
   ],
   "source": [
    "#####################\n",
    "## Motifs\n",
    "## see http://biopython.readthedocs.io/en/latest/Tutorial/chapter_motifs.html#position-specific-scoring-matrices\n",
    "#####################\n",
    "\n",
    "pwm = motif.counts.normalize(pseudocounts=0.0) # Doesn't change from pwm\n",
    "print(pwm)\n",
    "pssm = pwm.log_odds()\n",
    "\n",
    "motif_length = len(motif) #for later retrival of nucleotide sequence\n",
    "\n",
    "######################\n",
    "## Searching for Motifs in Sequences\n",
    "######################\n",
    "\n",
    "## Returns a list of arrays with a score for each position\n",
    "## This give the score for each position\n",
    "## If you print the length you get the length of the sequence minus TFBS length. \n",
    "\n",
    "pssm_list = [ ]\n",
    "for seq in raw_sequences_2:\n",
    "    pssm_list.append(pssm.calculate(seq))\n",
    "\n",
    "## At this point there is a score for each position - length of motif\n",
    "print(len(raw_sequences_2[6]))\n",
    "print(len(pssm_list[6]))\n",
    "print(pssm_list[6])\n",
    "\n",
    "########################## \n",
    "## Automatic Calculation of threshold\n",
    "##########################\n",
    "\n",
    "## Ideal to find something that automatically calculates, as\n",
    "## opposed to having human choosing threshold\n",
    "\n",
    "## Approximate calculation of appropriate thresholds for motif finding \n",
    "## Patser Threshold\n",
    "## It selects such a threshold that the log(fpr)=-ic(M) \n",
    "## note: the actual patser software uses natural logarithms instead of log_2, so the numbers \n",
    "## are not directly comparable. \n",
    "\n",
    "## distribution = pssm.distribution(background=motif.background, precision=10**4)\n",
    "## patser_threshold = distribution.threshold_patser() #for use later\n",
    "\n",
    "## for sys.out\n",
    "## print(\"Patser Threshold is %5.3f\" % patser_threshold) # Calculates Paster threshold. \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3077\n",
      "[3291, 3518, 3054, 3670, 2926, 3695, 3077, 3127, 2823, 2824, 2530, 2621, 2591, 2900, 1680, 3314, 3095, 3306, 2986, 3006, 3292, 2491, 2840, 2631]\n",
      "['VT11157|1|MEMB002B|-|3291', 'VT11157|1|MEMB004A|+|3518', 'VT11157|1|MEMB007B|-|3054', 'VT11157|1|MEMB002C|-|3670', 'VT11157|1|MEMB003C|-|2926', 'VT11157|1|MEMB002A|+|3695', 'VT11157|1|MEMB002F|-|3077', 'VT11157|1|MEMB007D|-|3127', 'VT11157|1|MEMB007C|+|2823', 'VT11157|1|MEMB003B|+|2824', 'VT11157|1|MEMB006B|+|2530', 'VT11157|1|MEMB003F|+|2621', 'VT11157|1|MEMB002E|+|2591', 'VT11157|1|MEMB002D|+|2900', 'VT11157|1|MEMB005D|-|1680', 'VT11157|1|MEMB006A|-|3314', 'VT11157|1|dkik|-|3095', 'VT11157|1|MEMB003A|-|3306', 'VT11157|1|MEMB003D|-|2986', 'VT11157|1|MEMB004E|-|3006', 'VT11157|1|MEMB004B|-|3292', 'VT11157|1|MEMB005B|-|2491', 'VT11157|1|MEMB006C|+|2840', 'VT11157|1|MEMB008C|-|2631']\n",
      "24\n"
     ]
    }
   ],
   "source": [
    "###################################\n",
    "## Searching for motif in all raw_sequences\n",
    "#################################\n",
    "\n",
    "raw_id = []\n",
    "for seq in raw_sequences:\n",
    "    raw_id.append(seq.id)\n",
    "\n",
    "record_length = []\n",
    "for record in raw_sequences_2:\n",
    "    record_length.append(len(record))\n",
    " \n",
    "print(len(raw_sequences_2[6])) \n",
    "# print(raw_sequences_2[6])\n",
    "print(record_length)\n",
    "print(raw_id)\n",
    "\n",
    "test = pssm.calculate(raw_sequences_2[6])\n",
    "test = pd.DataFrame(test)\n",
    "#print(test)\n",
    "\n",
    "## The problem is that calculate and search are different and without a threshold, search acts weird.\n",
    "## This gives you an array of positions. \n",
    "score = []\n",
    "for seq in raw_sequences_2:\n",
    "    counter = 0\n",
    "    score.append(pssm.calculate(raw_sequences_2[counter]))\n",
    "    counter + 1 \n",
    "    \n",
    "print(len(score))\n",
    "    \n",
    "# for position, score in pssm.search(raw_sequences_2[6]):\n",
    "#         positions = {'species': raw_id[i], 'score':score, 'position':position, 'seq_len': record_length[i] }\n",
    "#         position_list.append(positions)\n",
    "\n",
    "# #print(pd.DataFrame(position_list))\n",
    "# df = pd.DataFrame(position_list)\n",
    "# print(df.shape)\n",
    "# print(df.sort_values(by=['species','position'], ascending=[True, False]))\n",
    "\n",
    "# print((3077 * 24/2))\n",
    "        \n",
    "## position_list = []\n",
    "## for i in range(len(raw_id)):\n",
    "##    for position, score in pssm.search(raw_sequences_2[i]):\n",
    "##        positions = {'species': raw_id[i], 'score':score, 'position':position, 'seq_len': record_length[i] }\n",
    "##        position_list.append(positions)\n",
    "\n",
    "## position_DF = pd.DataFrame(position_list)\n",
    "## position_DF = position_DF.sort_values(by=['species','position'], ascending=[True, False])\n",
    "## print(position_DF)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       position     score  seq_len                    species\n",
      "40270      3691 -6.201628     3695  VT11157|1|MEMB002A|+|3695\n",
      "40268      3690 -4.140227     3695  VT11157|1|MEMB002A|+|3695\n",
      "40266      3689 -6.140227     3695  VT11157|1|MEMB002A|+|3695\n",
      "40264      3688 -5.140227     3695  VT11157|1|MEMB002A|+|3695\n",
      "40262      3687 -8.403262     3695  VT11157|1|MEMB002A|+|3695\n",
      "40260      3686 -1.953700     3695  VT11157|1|MEMB002A|+|3695\n",
      "40258      3685 -6.555265     3695  VT11157|1|MEMB002A|+|3695\n",
      "40256      3684 -2.307337     3695  VT11157|1|MEMB002A|+|3695\n",
      "40254      3683 -3.140227     3695  VT11157|1|MEMB002A|+|3695\n",
      "40252      3682 -3.345811     3695  VT11157|1|MEMB002A|+|3695\n",
      "40250      3681 -1.193808     3695  VT11157|1|MEMB002A|+|3695\n",
      "40248      3680 -1.953700     3695  VT11157|1|MEMB002A|+|3695\n",
      "40246      3679  1.007825     3695  VT11157|1|MEMB002A|+|3695\n",
      "40244      3678 -3.307337     3695  VT11157|1|MEMB002A|+|3695\n",
      "40242      3677 -3.140227     3695  VT11157|1|MEMB002A|+|3695\n",
      "40240      3676 -7.555265     3695  VT11157|1|MEMB002A|+|3695\n",
      "40238      3675 -1.345811     3695  VT11157|1|MEMB002A|+|3695\n",
      "40236      3674 -4.201628     3695  VT11157|1|MEMB002A|+|3695\n",
      "40234      3673  0.069226     3695  VT11157|1|MEMB002A|+|3695\n",
      "40232      3672 -3.345811     3695  VT11157|1|MEMB002A|+|3695\n",
      "40230      3671 -3.140227     3695  VT11157|1|MEMB002A|+|3695\n",
      "40228      3670 -6.140227     3695  VT11157|1|MEMB002A|+|3695\n",
      "40226      3669 -3.193808     3695  VT11157|1|MEMB002A|+|3695\n",
      "40224      3668 -1.953700     3695  VT11157|1|MEMB002A|+|3695\n",
      "40222      3667 -4.201628     3695  VT11157|1|MEMB002A|+|3695\n",
      "40220      3666 -0.892300     3695  VT11157|1|MEMB002A|+|3695\n",
      "40218      3665 -5.140227     3695  VT11157|1|MEMB002A|+|3695\n",
      "40216      3664 -5.140227     3695  VT11157|1|MEMB002A|+|3695\n",
      "40214      3663 -8.403262     3695  VT11157|1|MEMB002A|+|3695\n",
      "40212      3662 -1.953700     3695  VT11157|1|MEMB002A|+|3695\n",
      "...         ...       ...      ...                        ...\n",
      "95245     -3066 -6.140227     3095      VT11157|1|dkik|-|3095\n",
      "95243     -3067 -4.140227     3095      VT11157|1|dkik|-|3095\n",
      "95241     -3068 -1.953700     3095      VT11157|1|dkik|-|3095\n",
      "95239     -3069  1.069226     3095      VT11157|1|dkik|-|3095\n",
      "95237     -3070  3.255753     3095      VT11157|1|dkik|-|3095\n",
      "95235     -3071 -5.403262     3095      VT11157|1|dkik|-|3095\n",
      "95233     -3072 -6.555265     3095      VT11157|1|dkik|-|3095\n",
      "95231     -3073 -3.307337     3095      VT11157|1|dkik|-|3095\n",
      "95229     -3074 -5.201628     3095      VT11157|1|dkik|-|3095\n",
      "95227     -3075 -0.892300     3095      VT11157|1|dkik|-|3095\n",
      "95225     -3076 -3.193808     3095      VT11157|1|dkik|-|3095\n",
      "95223     -3077 -5.140227     3095      VT11157|1|dkik|-|3095\n",
      "95221     -3078 -4.201628     3095      VT11157|1|dkik|-|3095\n",
      "95219     -3079 -3.345811     3095      VT11157|1|dkik|-|3095\n",
      "95217     -3080 -0.930774     3095      VT11157|1|dkik|-|3095\n",
      "95215     -3081  1.069226     3095      VT11157|1|dkik|-|3095\n",
      "95213     -3082 -1.345811     3095      VT11157|1|dkik|-|3095\n",
      "95211     -3083 -2.307337     3095      VT11157|1|dkik|-|3095\n",
      "95209     -3084 -0.345811     3095      VT11157|1|dkik|-|3095\n",
      "95207     -3085 -4.307337     3095      VT11157|1|dkik|-|3095\n",
      "95205     -3086 -5.403262     3095      VT11157|1|dkik|-|3095\n",
      "95203     -3087 -8.555264     3095      VT11157|1|dkik|-|3095\n",
      "95201     -3088 -4.140227     3095      VT11157|1|dkik|-|3095\n",
      "95199     -3089 -0.992175     3095      VT11157|1|dkik|-|3095\n",
      "95197     -3090  0.069226     3095      VT11157|1|dkik|-|3095\n",
      "95195     -3091  0.046300     3095      VT11157|1|dkik|-|3095\n",
      "95193     -3092 -0.345811     3095      VT11157|1|dkik|-|3095\n",
      "95191     -3093 -0.059410     3095      VT11157|1|dkik|-|3095\n",
      "95189     -3094 -5.403262     3095      VT11157|1|dkik|-|3095\n",
      "95187     -3095  0.902116     3095      VT11157|1|dkik|-|3095\n",
      "\n",
      "[142432 rows x 4 columns]\n"
     ]
    }
   ],
   "source": [
    "\n",
    "raw_id = []\n",
    "for seq in raw_sequences:\n",
    "    raw_id.append(seq.id)\n",
    "\n",
    "record_length = []\n",
    "for record in raw_sequences_2:\n",
    "    record_length.append(len(record))\n",
    "\n",
    "position_list = []\n",
    "for i in range(len(raw_id)):\n",
    "    for position, score in pssm.search(raw_sequences_2[i], threshold = -50):\n",
    "        positions = {'species': raw_id[i], 'score':score, 'position':position, 'seq_len': record_length[i] }\n",
    "        position_list.append(positions)\n",
    "        \n",
    "position_DF = pd.DataFrame(position_list)\n",
    "position_DF = position_DF.sort_values(by=['species','position'], ascending=[True, False])\n",
    "print(position_DF)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#############################\n",
    "## Add strand and pos position information as columns to position_DF\n",
    "#############################\n",
    "\n",
    "position_list_pos = []\n",
    "for i, x in enumerate(position_DF['position']):\n",
    "    if x < 0:\n",
    "       position_list_pos.append(position_DF.loc[position_DF.index[i], 'seq_len'] + x)\n",
    "    else:\n",
    "       position_list_pos.append(x)\n",
    "\n",
    "## append to position_DF\n",
    "position_DF['raw_position'] = position_list_pos\n",
    "    \n",
    "## strand Column\n",
    "strand = []\n",
    "for x in position_DF['position']:\n",
    "    if x < 0:\n",
    "       strand.append(\"negative\")\n",
    "    else:\n",
    "       strand.append(\"positive\")\n",
    "    \n",
    "## append to position_DF\n",
    "position_DF['strand'] = strand\n",
    "\n",
    "## motif_found column\n",
    "raw_sequences_2_list = []\n",
    "for seq in raw_sequences_2:\n",
    "    raw_sequences_2_list.append(str(seq))\n",
    "    \n",
    "position_DF = position_DF.sort_values(by=['species','raw_position'], ascending=[True, False])\n",
    "position_DF.to_csv('map_motif_test.csv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       align_position  raw_position                    species\n",
      "16459               0             0  VT11157|1|MEMB002A|+|3695\n",
      "16460               1             1  VT11157|1|MEMB002A|+|3695\n",
      "16461               2             2  VT11157|1|MEMB002A|+|3695\n",
      "16462               3             3  VT11157|1|MEMB002A|+|3695\n",
      "16463               4             4  VT11157|1|MEMB002A|+|3695\n",
      "16464               5             5  VT11157|1|MEMB002A|+|3695\n",
      "16465               6             6  VT11157|1|MEMB002A|+|3695\n",
      "16466               7             7  VT11157|1|MEMB002A|+|3695\n",
      "16467               8             8  VT11157|1|MEMB002A|+|3695\n",
      "16468               9             9  VT11157|1|MEMB002A|+|3695\n",
      "16469              10            10  VT11157|1|MEMB002A|+|3695\n",
      "16470              11            11  VT11157|1|MEMB002A|+|3695\n",
      "16471              12            12  VT11157|1|MEMB002A|+|3695\n",
      "16472              13            13  VT11157|1|MEMB002A|+|3695\n",
      "16473              14            14  VT11157|1|MEMB002A|+|3695\n",
      "16474              15            15  VT11157|1|MEMB002A|+|3695\n",
      "16475              16            16  VT11157|1|MEMB002A|+|3695\n",
      "16476              17            17  VT11157|1|MEMB002A|+|3695\n",
      "16477              20            18  VT11157|1|MEMB002A|+|3695\n",
      "16478              21            19  VT11157|1|MEMB002A|+|3695\n",
      "16479              22            20  VT11157|1|MEMB002A|+|3695\n",
      "16480              23            21  VT11157|1|MEMB002A|+|3695\n",
      "16481              24            22  VT11157|1|MEMB002A|+|3695\n",
      "16482              25            23  VT11157|1|MEMB002A|+|3695\n",
      "16483              26            24  VT11157|1|MEMB002A|+|3695\n",
      "16484              27            25  VT11157|1|MEMB002A|+|3695\n",
      "16485              28            26  VT11157|1|MEMB002A|+|3695\n",
      "16486              29            27  VT11157|1|MEMB002A|+|3695\n",
      "16487              30            28  VT11157|1|MEMB002A|+|3695\n",
      "16488              31            29  VT11157|1|MEMB002A|+|3695\n",
      "...               ...           ...                        ...\n",
      "50706            5750          3065      VT11157|1|dkik|-|3095\n",
      "50707            5751          3066      VT11157|1|dkik|-|3095\n",
      "50708            5752          3067      VT11157|1|dkik|-|3095\n",
      "50709            5753          3068      VT11157|1|dkik|-|3095\n",
      "50710            5754          3069      VT11157|1|dkik|-|3095\n",
      "50711            5755          3070      VT11157|1|dkik|-|3095\n",
      "50712            5756          3071      VT11157|1|dkik|-|3095\n",
      "50713            5757          3072      VT11157|1|dkik|-|3095\n",
      "50714            5758          3073      VT11157|1|dkik|-|3095\n",
      "50715            5759          3074      VT11157|1|dkik|-|3095\n",
      "50716            5760          3075      VT11157|1|dkik|-|3095\n",
      "50717            5761          3076      VT11157|1|dkik|-|3095\n",
      "50718            5768          3077      VT11157|1|dkik|-|3095\n",
      "50719            5769          3078      VT11157|1|dkik|-|3095\n",
      "50720            5770          3079      VT11157|1|dkik|-|3095\n",
      "50721            5771          3080      VT11157|1|dkik|-|3095\n",
      "50722            5772          3081      VT11157|1|dkik|-|3095\n",
      "50723            5773          3082      VT11157|1|dkik|-|3095\n",
      "50724            5774          3083      VT11157|1|dkik|-|3095\n",
      "50725            5775          3084      VT11157|1|dkik|-|3095\n",
      "50726            5776          3085      VT11157|1|dkik|-|3095\n",
      "50727            5777          3086      VT11157|1|dkik|-|3095\n",
      "50728            5778          3087      VT11157|1|dkik|-|3095\n",
      "50729            5779          3088      VT11157|1|dkik|-|3095\n",
      "50730            5780          3089      VT11157|1|dkik|-|3095\n",
      "50731            5781          3090      VT11157|1|dkik|-|3095\n",
      "50732            5782          3091      VT11157|1|dkik|-|3095\n",
      "50733            5783          3092      VT11157|1|dkik|-|3095\n",
      "50734            5784          3093      VT11157|1|dkik|-|3095\n",
      "50735            5785          3094      VT11157|1|dkik|-|3095\n",
      "\n",
      "[71288 rows x 3 columns]\n"
     ]
    }
   ],
   "source": [
    "##################\n",
    "## get alignment position \n",
    "#################\n",
    "\n",
    "remap_list = []\n",
    "nuc_list = ['A', 'a', 'G', 'g', 'C', 'c', 'T', 't', 'N', 'n']\n",
    "\n",
    "for i in range(len(raw_id)):\n",
    "    counter = 0\n",
    "    for xInd, x in enumerate(alignment[i].seq):    \n",
    "        if x in nuc_list:\n",
    "            remaps = {'raw_position': counter, 'align_position':xInd, 'species':alignment[i].id}\n",
    "            counter += 1\n",
    "            remap_list.append(remaps)\n",
    "            \n",
    "remap_DF = pd.DataFrame(remap_list)\n",
    "remap_DF = remap_DF.sort_values(by=['species','raw_position'], ascending=[True, True])\n",
    "\n",
    "\n",
    "print(remap_DF)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        position     score  seq_len                    species  raw_position  \\\n",
      "0         3691.0 -6.201628   3695.0  VT11157|1|MEMB002A|+|3695          3691   \n",
      "1           -4.0 -0.193808   3695.0  VT11157|1|MEMB002A|+|3695          3691   \n",
      "2         3690.0 -4.140227   3695.0  VT11157|1|MEMB002A|+|3695          3690   \n",
      "3           -5.0 -0.059410   3695.0  VT11157|1|MEMB002A|+|3695          3690   \n",
      "4         3689.0 -6.140227   3695.0  VT11157|1|MEMB002A|+|3695          3689   \n",
      "5           -6.0 -1.155334   3695.0  VT11157|1|MEMB002A|+|3695          3689   \n",
      "6         3688.0 -5.140227   3695.0  VT11157|1|MEMB002A|+|3695          3688   \n",
      "7           -7.0  0.940590   3695.0  VT11157|1|MEMB002A|+|3695          3688   \n",
      "8         3687.0 -8.403262   3695.0  VT11157|1|MEMB002A|+|3695          3687   \n",
      "9           -8.0 -3.155334   3695.0  VT11157|1|MEMB002A|+|3695          3687   \n",
      "10        3686.0 -1.953700   3695.0  VT11157|1|MEMB002A|+|3695          3686   \n",
      "11          -9.0  1.054119   3695.0  VT11157|1|MEMB002A|+|3695          3686   \n",
      "12        3685.0 -6.555265   3695.0  VT11157|1|MEMB002A|+|3695          3685   \n",
      "13         -10.0 -7.403262   3695.0  VT11157|1|MEMB002A|+|3695          3685   \n",
      "14        3684.0 -2.307337   3695.0  VT11157|1|MEMB002A|+|3695          3684   \n",
      "15         -11.0 -3.140227   3695.0  VT11157|1|MEMB002A|+|3695          3684   \n",
      "16        3683.0 -3.140227   3695.0  VT11157|1|MEMB002A|+|3695          3683   \n",
      "17         -12.0 -3.307337   3695.0  VT11157|1|MEMB002A|+|3695          3683   \n",
      "18        3682.0 -3.345811   3695.0  VT11157|1|MEMB002A|+|3695          3682   \n",
      "19         -13.0 -3.201628   3695.0  VT11157|1|MEMB002A|+|3695          3682   \n",
      "20        3681.0 -1.193808   3695.0  VT11157|1|MEMB002A|+|3695          3681   \n",
      "21         -14.0 -1.953700   3695.0  VT11157|1|MEMB002A|+|3695          3681   \n",
      "22        3680.0 -1.953700   3695.0  VT11157|1|MEMB002A|+|3695          3680   \n",
      "23         -15.0 -2.193808   3695.0  VT11157|1|MEMB002A|+|3695          3680   \n",
      "24        3679.0  1.007825   3695.0  VT11157|1|MEMB002A|+|3695          3679   \n",
      "25         -16.0 -0.992175   3695.0  VT11157|1|MEMB002A|+|3695          3679   \n",
      "26        3678.0 -3.307337   3695.0  VT11157|1|MEMB002A|+|3695          3678   \n",
      "27         -17.0 -3.140227   3695.0  VT11157|1|MEMB002A|+|3695          3678   \n",
      "28        3677.0 -3.140227   3695.0  VT11157|1|MEMB002A|+|3695          3677   \n",
      "29         -18.0 -2.307337   3695.0  VT11157|1|MEMB002A|+|3695          3677   \n",
      "...          ...       ...      ...                        ...           ...   \n",
      "142474       NaN       NaN      NaN  VT11157|1|MEMB005B|-|2491          2488   \n",
      "142475       NaN       NaN      NaN  VT11157|1|MEMB005B|-|2491          2489   \n",
      "142476       NaN       NaN      NaN  VT11157|1|MEMB005B|-|2491          2490   \n",
      "142477       NaN       NaN      NaN  VT11157|1|MEMB005D|-|1680          1677   \n",
      "142478       NaN       NaN      NaN  VT11157|1|MEMB005D|-|1680          1678   \n",
      "142479       NaN       NaN      NaN  VT11157|1|MEMB005D|-|1680          1679   \n",
      "142480       NaN       NaN      NaN  VT11157|1|MEMB006A|-|3314          3311   \n",
      "142481       NaN       NaN      NaN  VT11157|1|MEMB006A|-|3314          3312   \n",
      "142482       NaN       NaN      NaN  VT11157|1|MEMB006A|-|3314          3313   \n",
      "142483       NaN       NaN      NaN  VT11157|1|MEMB006B|+|2530          2527   \n",
      "142484       NaN       NaN      NaN  VT11157|1|MEMB006B|+|2530          2528   \n",
      "142485       NaN       NaN      NaN  VT11157|1|MEMB006B|+|2530          2529   \n",
      "142486       NaN       NaN      NaN  VT11157|1|MEMB006C|+|2840          2837   \n",
      "142487       NaN       NaN      NaN  VT11157|1|MEMB006C|+|2840          2838   \n",
      "142488       NaN       NaN      NaN  VT11157|1|MEMB006C|+|2840          2839   \n",
      "142489       NaN       NaN      NaN  VT11157|1|MEMB007B|-|3054          3051   \n",
      "142490       NaN       NaN      NaN  VT11157|1|MEMB007B|-|3054          3052   \n",
      "142491       NaN       NaN      NaN  VT11157|1|MEMB007B|-|3054          3053   \n",
      "142492       NaN       NaN      NaN  VT11157|1|MEMB007C|+|2823          2820   \n",
      "142493       NaN       NaN      NaN  VT11157|1|MEMB007C|+|2823          2821   \n",
      "142494       NaN       NaN      NaN  VT11157|1|MEMB007C|+|2823          2822   \n",
      "142495       NaN       NaN      NaN  VT11157|1|MEMB007D|-|3127          3124   \n",
      "142496       NaN       NaN      NaN  VT11157|1|MEMB007D|-|3127          3125   \n",
      "142497       NaN       NaN      NaN  VT11157|1|MEMB007D|-|3127          3126   \n",
      "142498       NaN       NaN      NaN  VT11157|1|MEMB008C|-|2631          2628   \n",
      "142499       NaN       NaN      NaN  VT11157|1|MEMB008C|-|2631          2629   \n",
      "142500       NaN       NaN      NaN  VT11157|1|MEMB008C|-|2631          2630   \n",
      "142501       NaN       NaN      NaN      VT11157|1|dkik|-|3095          3092   \n",
      "142502       NaN       NaN      NaN      VT11157|1|dkik|-|3095          3093   \n",
      "142503       NaN       NaN      NaN      VT11157|1|dkik|-|3095          3094   \n",
      "\n",
      "          strand  align_position  \n",
      "0       positive            5782  \n",
      "1       negative            5782  \n",
      "2       positive            5781  \n",
      "3       negative            5781  \n",
      "4       positive            5780  \n",
      "5       negative            5780  \n",
      "6       positive            5779  \n",
      "7       negative            5779  \n",
      "8       positive            5778  \n",
      "9       negative            5778  \n",
      "10      positive            5777  \n",
      "11      negative            5777  \n",
      "12      positive            5776  \n",
      "13      negative            5776  \n",
      "14      positive            5775  \n",
      "15      negative            5775  \n",
      "16      positive            5774  \n",
      "17      negative            5774  \n",
      "18      positive            5773  \n",
      "19      negative            5773  \n",
      "20      positive            5772  \n",
      "21      negative            5772  \n",
      "22      positive            5771  \n",
      "23      negative            5771  \n",
      "24      positive            5770  \n",
      "25      negative            5770  \n",
      "26      positive            5769  \n",
      "27      negative            5769  \n",
      "28      positive            5768  \n",
      "29      negative            5768  \n",
      "...          ...             ...  \n",
      "142474       NaN            5783  \n",
      "142475       NaN            5784  \n",
      "142476       NaN            5785  \n",
      "142477       NaN            5783  \n",
      "142478       NaN            5784  \n",
      "142479       NaN            5785  \n",
      "142480       NaN            5783  \n",
      "142481       NaN            5784  \n",
      "142482       NaN            5785  \n",
      "142483       NaN            5784  \n",
      "142484       NaN            5785  \n",
      "142485       NaN            5786  \n",
      "142486       NaN            5497  \n",
      "142487       NaN            5498  \n",
      "142488       NaN            5499  \n",
      "142489       NaN            5783  \n",
      "142490       NaN            5784  \n",
      "142491       NaN            5785  \n",
      "142492       NaN            5783  \n",
      "142493       NaN            5784  \n",
      "142494       NaN            5785  \n",
      "142495       NaN            5783  \n",
      "142496       NaN            5784  \n",
      "142497       NaN            5785  \n",
      "142498       NaN            5783  \n",
      "142499       NaN            5784  \n",
      "142500       NaN            5785  \n",
      "142501       NaN            5783  \n",
      "142502       NaN            5784  \n",
      "142503       NaN            5785  \n",
      "\n",
      "[142504 rows x 7 columns]\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'alignment_file_name' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-110-2ef629e611a3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     10\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m \u001b[0;31m## Attach File Name\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 12\u001b[0;31m \u001b[0mTFBS_map_DF_all\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'alignment_file'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0malignment_file_name\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     13\u001b[0m \u001b[0mTFBS_map_DF_all\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'motif_file'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmotif_file_name\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     14\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'alignment_file_name' is not defined"
     ]
    }
   ],
   "source": [
    "## Merge both datasets\n",
    "## Merge - all sites\n",
    "\n",
    "TFBS_map_DF_all = pd.merge(position_DF, remap_DF, on=['species', 'raw_position'], how='outer')\n",
    "\n",
    "print(TFBS_map_DF_all)\n",
    "\n",
    "## Sort\n",
    "TFBS_map_DF_all = TFBS_map_DF_all.sort_values(by=['species','align_position'], ascending=[True, True])\n",
    "\n",
    "## Attach File Name\n",
    "TFBS_map_DF_all['alignment_file'] = alignment_file_name\n",
    "TFBS_map_DF_all['motif_file'] = motif_file_name\n",
    "\n",
    "## Remove NAs\n",
    "TFBS_map_DF_all = TFBS_map_DF_all.dropna()\n",
    "\n",
    "##remove values under the threshold if there is one\n",
    "print (\"Output Threshold: \" + str(threshold))\n",
    "TFBS_map_DF_all = TFBS_map_DF_all.loc[TFBS_map_DF_all['score'] >= float(threshold)] \n",
    "\n",
    "## Write out Files\n",
    "# print (\"Success!!\")\n",
    "# print (\"The output csv file is located at: \" + os.getcwd())\n",
    "# print (\"Under the name: map_motif\" + alignment_file_name + \"-\" + motif_file_name + \".csv\")\n",
    "TFBS_map_DF_all.to_csv('map_motif_test.csv', sep='\\t', na_rep=\"NA\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
